### ISSUE TITLE
How to use custom models to active learning

### ISSUE THREAD

**Issue by CastleDream** [microsoft/VoTT#834]
I have tried many many times to use tensorflowjs_converter to change my tensorflow.pd to a model.json
while I still can't figure out the specific format of model.json and classes.json in the example of cocoSSDModel which is the pre-trained model in Vott' active learning.

I want to know which method do you adopt to convert and get the model.json and classes.json
which one of TensorFlow SavedModel, TensorFlow Hub module, Keras HDF5 or tf.keras SavedModel???
and also, I noticed that the format of model.json of mine is different from the given, even I have the 
same version with the Vott，i.e. tensorflowjs=1.0.3.

can someone give me a complete and detailed example of a custom model in active learning of Vott

**Comment by ghunkins** [comment]
+1 on this. Excellent tool, would love custom active learning.

**Comment by nansravn** [comment]
+1 on this. I'm trying to use a custom vision model that has been exported as a docker image. I have this model exported on the http://127.0.0.1/image path, but I can't use it for active learning. =/

**Comment by JacopoMangiavacchi** [comment]
@nansravn you can't reuse a custom vision model exported as a docker image.  You need to convert the model to the TF.js format needed by VoTT.

Once you have your model saved in the TF.js format put the model.json file, the weights binary files referenced in the model.json file and the custom classes.json file containing your labels on a folder and point to that folder from VoTT Active Learning settings. Please look at the cocoSSDModel on this repo for a sample of all of these files.

#829

**Comment by naveenchary121** [comment]
>i need help in creating vott active learning model plzzz help me

**Comment by murali513** [comment]
I want to know which method do you adopt to convert and get the model.json and classes.json
which one of TensorFlow SavedModel, TensorFlow Hub module, Keras HDF5 or tf.keras SavedModel???

**Comment by RafayAK** [comment]
> @nansravn you can't reuse a custom vision model exported as a docker image. You need to convert the model to the TF.js format needed by VoTT.
> 
> Once you have your model saved in the TF.js format put the model.json file, the weights binary files referenced in the model.json file and the custom classes.json file containing your labels on a folder and point to that folder from VoTT Active Learning settings. Please look at the cocoSSDModel on this repo for a sample of all of these files.
> 
> #829

Where do I put the pre and post-processing steps?

**Comment by ghunkins** [comment]
@RafayAK An option may be to look into using a `tf.keras.layers.Lambda` layer.

**Comment by RafayAK** [comment]
@ghunkins I'm not sure I can do the pre and post-processing steps of a Keras Retina-Net-like model in lambda layers and successfully convert to Tensorflow Js. Do you have any resources or references? 
I have an open issue on this #963
